zipline/pipeline/loaders/utils.py

```diff
@@ -36,28 +42,35 @@ def next_date_frame(dates, events_by_sid, event_date_field_name):
     --------
     previous_date_frame
     """
-    cols = {
+    date_cols = {
         equity: np.full_like(dates, NaTns) for equity in events_by_sid
     }
+    value_cols = {
+        equity: np.full(len(dates), missing_value, dtype=field_dtype)
+        for equity in events_by_sid
+    }
+
     raw_dates = dates.values
     for equity, df in iteritems(events_by_sid):
         event_dates = df[event_date_field_name]
-        data = cols[equity]
+        values = df[return_field_name]
+        data = date_cols[equity]
         if not event_dates.index.is_monotonic_increasing:
             event_dates = event_dates.sort_index()
 
         # Iterate over the raw Series values, since we're comparing against
         # numpy arrays anyway.
-        iterkv = zip(event_dates.index.values, event_dates.values)
-        for knowledge_date, event_date in iterkv:
+        iterkv = zip(event_dates.index.values, event_dates.values, values)
```

Comment:
idk if we want to keep calling this iterkv because it is an iterator of three-tuples.